src/vue/common/CollectionLoader.spec.js

```diff
@@ -0,0 +1,209 @@
+import CollectionLoader from './CollectionLoader'
+import { shallowMount, createLocalVue } from '@vue/test-utils'
+import { globalize } from '@/vue/filters/globalize'
+import { ErrorHandler } from '@/js/helpers/error-handler'
+import { eventHandlerTest } from '@/js/helpers/test-helper'
+
+const localVue = createLocalVue()
+localVue.filter('globalize', globalize)
+
+describe('CollectionLoader component test', () => {
+  const firstPageLoader = sinon.stub().resolves({
+    data: [],
+    fetchNext: () => {}
+  })
+  afterEach(() => {
+    sinon.restore()
+  })
+
+  it('loadPage calls provided loaderFn', async () => {
+    const wrapper = shallowMount(CollectionLoader, {
+      propsData: {
+        firstPageLoader
+      },
+      localVue
+    })
+    const spy = sinon.stub().resolves({
+      data: [],
+      fetchNext: () => {}
+    })
+
+    await wrapper.vm.loadPage('some-event', spy)
+
+    expect(spy.calledOnce).to.be.true
+  })
+
+  it('loadPage properly emits event with correct payload', async () => {
+    const wrapper = shallowMount(CollectionLoader, {
+      propsData: {
+        firstPageLoader
+      },
+      localVue
+    })
+    const mockData = []
```

Comment:
if we call it `data` we can do `.resolves({ data })`

Reply:
ok